flink-yarn/src/test/java/org/apache/flink/yarn/FlinkYarnSessionCliTest.java

```diff
@@ -137,6 +137,32 @@ public void testCorrectSettingOfDetachedMode() throws Exception {
 		assertTrue(descriptor.isDetachedMode());
 	}
 
+	@Test
+	public void testConfigurationBeforeDeployment() throws Exception {
+		String[] params = new String[] { "-yD", "yarn.application.priority=10"};
+
+		FlinkYarnSessionCli yarnCLI = new FlinkYarnSessionCli(
+			new Configuration(),
+			tmp.getRoot().getAbsolutePath(),
+			"y",
+			"yarn");
+
+		CommandLine commandLine = yarnCLI.parseCommandLineOptions(params, true);
+
+		AbstractYarnClusterDescriptor descriptor = yarnCLI.createClusterDescriptor(commandLine);
+		Configuration flinkConfiguration = descriptor.getFlinkConfiguration();
+
+		// YarnClusterDescriptor will add dynamic properties to local flinkConfiguraton before deployment
+		Map<String, String> dynProperties =
+			FlinkYarnSessionCli.getDynamicProperties(descriptor.getDynamicPropertiesEncoded());
+		for (Map.Entry<String, String> dynProperty : dynProperties.entrySet()) {
+			flinkConfiguration.setString(dynProperty.getKey(), dynProperty.getValue());
+		}
+
+		int priority = flinkConfiguration.getInteger(YarnConfigOptions.APPLICATION_PRIORITY);
+		assertEquals(10, priority);
+	}
```

Comment:
What do we test here exactly? Shouldn't we rather test that given `YarnConfigOptions.APPLICATION_PRIORITY` being set that we create an `ApplicationSubmissionContext` with this priority?

Reply:
Yes, we could do something similar to the `AbstractYarnClusterTest` where we use the `TestingYarnClient` as a mock instance.